src/workerd/io/actor-cache.c++

```diff
@@ -2139,6 +2142,37 @@ void ActorCache::ensureFlushScheduled(const WriteOptions& options) {
   }
 }
 
+kj::Maybe<kj::Promise<void>> ActorCache::onNoPendingFlush() {
+  // This function returns a Maybe<Promise> because a falsy maybe allows the jsg interface to make
+  // a resolved jsg::Promise. This is meaningfully different from a ready kj::Promise because it
+  // allows the next continuation to run immediately on the microtask queue instead of returning to
+  // the kj event loop and fulfilling a resolver that enqueues the continuation.
+
+  if (lru.options.neverFlush) {
+    // We won't ever flush (usually because we're a preview session), so return a falsy maybe.
+    return nullptr;
+  }
+
+  if (flushScheduled) {
+    // Our current run has done operations that need to be flushed, we need to wait for that flush
```

Comment:
FWIW I don't think this comment is strictly accurate either -- it isn't necessarily our current run that set `flushScheduled` to true. Some other request could have done that, and the field just hasn't been set back to false yet because there was some previous flush that was still in progress when the other request scheduled an additional flush. In that case the previous flush would be keeping `lastFlush` from resolving and running the callback that would set `flushScheduled` to false for the new flush.
.
Given how hard it is to explain, I'm not sure this level of detail in the comments to distinguish between `flushScheduled` and `flushEnqueued` is all that useful when someone can look up above at how they're used. I'm not even totally sure how to explain it at this point, but I think my best attempt would be to rename `flushesEnqueued` to `flushInProgress` or something like that?

But you can keep these comments around if you want.

Reply:
Alright, I get what you're saying. Yeah, I'll trim the comment down to the bare minimum to explain the difference between the two conditions.